bpf/process/data_event.h

```diff
@@ -6,30 +6,26 @@
 static inline __attribute__((always_inline)) long
 __do_bytes(void *ctx, struct msg_data *msg, unsigned long uptr, size_t bytes)
 {
-	size_t max = sizeof(msg->arg) - 1;
-	size_t rd_bytes, size;
 	int err;
 
-	rd_bytes = bytes > max ? max : bytes;
-	msg->common.size = offsetof(struct msg_data, arg) + rd_bytes;
-
 	/* Code movement from clang forces us to inline bounds checks here */
-	asm volatile("%[rd_bytes] &= 0x7fff;\n"
-		     "if %[rd_bytes] < 32736 goto +1\n;"
-		     "%[rd_bytes] = 32736;\n"
-		     :
-		     : [rd_bytes] "+r"(rd_bytes)
-		     :);
-	err = probe_read(&msg->arg[0], rd_bytes, (char *)uptr);
+	asm volatile goto("%[bytes] &= 0x7fff;\n"
```

Comment:
I'm reading the comment in: https://github.com/cilium/tetragon/blob/38d3a6135cf9e23265c4319c3507bfffb3b42a5f/bpf/lib/data_msg.h#L18-L28

which describes the `size &= 0x7fff` part. Is this still needed after this change?

Reply:
Yes, you are correct. It seems that we can remove the `"%[bytes] &= 0x7fff;\n"` command as well. I will update the PR.

Reply:
Thanks!

How about we also add a:

```
#define MESSAGE_DATA_ARG_LEN 32736
```

And use that both in `struct msg_data` and the assembly code? (I think there should be a way to pass constants as arguments).

Reply:
Done and tested in 4.19, 5.4, 5.10, and 5.13.